pennylane/transforms/qcut.py

```diff
@@ -210,3 +213,235 @@ def tape_to_graph(tape: QuantumTape) -> MultiDiGraph:
             order += 1
 
     return graph
+
+
+@dataclass(frozen=True)
+class CutConfig:
+    """
+    Container wrapping partitioning parameters for passing to the automatic cutter.
+    """
+
+    k: int
+    imbalance: float
+    fragment_wires: List[int] = None
+    fragment_gates: List[int] = None
+
+
+def _infer_cut_configs(
+    num_tape_wires,
+    num_tape_gates,
+    max_device_wires,
+    max_device_gates,
+    min_device_wires=None,
+    min_device_gates=None,
+    k_lower=None,
+    k_upper=None,
+    imbalance_tolerance=None,
+    fragment_wires=None,
+    fragment_gates=None,
+):
+    """
+    Helper function for deriving the best default partitioning parameters for the automatic
+    circuit cutter. Arguments can be either user provided or inferred from devices.
+
+    Args:
+        num_tape_wires (int): Number of wires in the circuit tape to be partitioned.
+        num_tape_gates (int): Number of gates in the circuit tape to be partitioned.
+        max_device_wires (int): Number of wires for the largest available device.
+        max_device_gates (int): Maximum allowed circuit depth for the deepest available device.
```

Comment:
Good point! I think this can safely default to `num_tape_gates` which is basically to impose no gate constraint. `min_device_gates`, when not supplied, defaults to `max_device_gates` which is to say there's only one device size.